README.md

```diff
@@ -1,22 +1,134 @@
+     ```
+
+1. **Set Up Algolia**:
+   - Go to [Algolia](https://www.algolia.com/) and create a free account.
+   - After creating an account, create an Algolia app.
+   - Update your `.env/local` file with the following keys from your Algolia app:
+     ```plaintext
+     DJANGO_ALGOLIA_API_KEY=<your_algolia_api_key>
+     DJANGO_ALGOLIA_APPLICATION_ID=<your_algolia_application_id>
+     ```
+   - Ensure that your API key has index write permissions. You can ignore any onboarding wizard instructions provided by Algolia.
+
+1. **Run the Application**:
+   - In your terminal, navigate to the project directory and run the following command to start the application:
+     ```bash
+     make run
+     ```
+   - Leave this terminal session running and wait until you see that [Nest local](http://localhost:8000/api/v1) is responding.
+
+1. **Load Initial Data**:
+   - Open a new terminal session and run the following command to populate the database with initial data from fixtures:
+     ```bash
+     make load-data
+     ```
+
+1. **Index Data**:
+   - In the same terminal session, run the following command to index the data:
+     ```bash
+     make index-data
+     ```
+
+1. **Verify API Endpoints**:
+   - Check that the data is available via these API endpoints:
+     - [Projects Endpoint](http://localhost:8000/api/v1/owasp/search/project)
+     - [Issues Endpoint](http://localhost:8000/api/v1/owasp/search/issue)
+
+### Optional Steps (for managing or fetching data):
+
+If you plan to manage or fetch data, follow these additional steps:
+
+1. **Create a Super User**:
+   - Run the following command to create a super user for accessing the admin interface:
+     ```bash
+     make setup
+     ```
+
+1. **Generate a GitHub Personal Access Token**:
+   - Create a GitHub [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+1. **Update Environment Variables with GitHub Token**:
+   - Open `backend/.env/local` again and update it with your GitHub token:
+     ```plaintext
+     GITHUB_TOKEN=<your_github_token>
+     ```
+
+1. **Sync Local Database Data**:
+   - Now you should be able to run the following command to sync your local database data with GitHub:
+     ```bash
+     make sync
+     ```
+
+
+## Linting and Formatting Setup
+
+To ensure code quality, we utilize ESLint and Prettier with Husky for pre-commit hooks.
+
+### Installing Dependencies
+
+Make sure you have the necessary dependencies installed:
+
+```bash
+npm install --save-dev eslint prettier husky lint-staged eslint-plugin-react eslint-plugin-react-hooks @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier eslint-plugin-prettier
+```
+### Configure Husky
+To set up Husky for pre-commit hooks, run:
+
+```bash
+npm run prepare
+```
+
+### Pre-commit Hook Script
+Ensure the `.husky/pre-commit` file consists of the following content:
+
+```bash
```

Comment:
Would it be a good idea to put it to the `.husky/pre-commit` file in the code base instead of creating it locally for everyone?

Reply:
@arkid15r  umm, I had problem setting up husky with the commands you initially suggested in the issue, as they were deprecated. I'm not sure if putting the husky inside the code base would be the best choice as versions might change again. Can we ask for reviews from the other collaborator aswell?

Reply:
Can we use `npx husky add .husky/pre-commit "npm run lint"` this one creates the husky pre-commit hook.

Reply:
@harsh3dev running `npm run prepare` would automatically install husky and create `.husky/pre-commit` hook with the required contents

<img width="1440" alt="Screenshot 2024-12-03 at 7 35 02 AM" src="https://github.com/user-attachments/assets/1a20811b-1433-4fb8-b39e-5424237a625b">
<img width="1440" alt="Screenshot 2024-12-03 at 7 35 24 AM" src="https://github.com/user-attachments/assets/9fcbc6cb-af16-4c63-a1fd-e56747f7b466">

@arkid15r this is how Linting structure/commands will be displayed `README.md`